src/completion/tab_handler.rs

```diff
@@ -50,15 +50,14 @@ impl ComplationActionHandler for DefaultCompletionActionHandler {
     //  in the line_buffer only the specified range of characters.
     // If internal index is 0 it means that is the first tab event pressed.
     // If internal index is greater than completions vector, we bring it back to 0.
-    fn handle(&mut self, line: &mut LineBuffer) {
+    fn handle(&mut self, present_buffer: &mut LineBuffer) {
+        // I am not sure what this block of code is trying to do
+        // We set a to b on first call then on all subsequent calls we set this value back (as on
+        // all calls this is the same value. No?)
         if self.index == 0 {
-            self.initial_line = LineBuffer::new();
-            self.initial_line.set_buffer(line.get_buffer().into());
-            self.initial_line
-                .set_insertion_point(line.line(), line.offset());
+            self.initial_line = present_buffer.clone();
         } else {
-            line.set_buffer(self.initial_line.get_buffer().into());
-            line.set_insertion_point(self.initial_line.line(), self.initial_line.offset())
+            *present_buffer = self.initial_line.clone();
         }
```

Comment:
@mzanrosso Can you help me understand this logic better. Why do we need to do this?